feat(nuxt): add useAsyncRouteData composable (#35820)
nuxt/nuxt: docs/4.api/2.composables/use-async-data.md

</script>
```

> ::tip
> For data that should always be scoped to the current page path (and receive `route` in the handler), use `useAsyncRouteData`.
> ::

Reviewer: ```suggestion
::tip
You can use `useAsyncRouteData` for fetching data that is scoped to the current page.
::
```

IMO it's not necessary to know that a composable, which the user reading this page about `useAsyncData` might know nothing about, receives the `route` in its handler. this sounds LLM generated.

Author: > this sounds LLM generated.

Sorry, but I am not a native speaker and I resort to the help of translators and LLM
